components/file_upload/file_upload.jsx

```diff
@@ -451,7 +450,7 @@ export class FileUpload extends PureComponent {
         }
 
         const target = this.props.getTarget();
-        const textarea = ReactDOM.findDOMNode(target);
+        const textarea = target?.wrapper?.current;
```

Comment:
To avoid this component needing to know that the `Textbox` contains has its own `wrapper` ref, could we instead have it so that the components that provide the `getTarget` prop do that extra step for us? They already need to know about some of the inner details of the `Textbox`, so I think it's good to keep all that encapsulation-breaking logic there without having it spread here.

Also, when you do that, is it possible to replace the `textbox.wrapper.current` with `textbox.getInputBox()`? Those are different DOM nodes, but I think that shouldn't matter too much. That'll help the keep the Textbox's internal details better hidden as well.

Reply:
I made the change.